drivers/storage/portworx/component/windows.go

```diff
@@ -0,0 +1,219 @@
+	if err != nil {
+		return false
+	}
+
+	w.isWindowsNode = isWindowsNode(nodeList)
+	return w.isWindowsNode
+}
+
+func (w *windows) Reconcile(cluster *corev1.StorageCluster) error {
+	ownRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+
+	var errList []string
+
+	if err := w.createDaemonSet(WindowsDaemonSetFileName, ownRef, cluster); err != nil {
+		logrus.Errorf("error during creating %s daemonset %s ", WindowsDaemonSetName, err)
+		errList = append(errList, err.Error())
+	}
+
+	if err := w.createStorageClass(); err != nil {
+		logrus.Errorf("error during creating %s storageclass %s ", WindowsStorageClass, err)
+		errList = append(errList, err.Error())
+	}
+
+	if len(errList) > 0 {
+		return commonerrors.New(strings.Join(errList, " , "))
+	}
+	return nil
+}
+
+func (w *windows) Delete(cluster *corev1.StorageCluster) error {
+
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+	if err := k8s.DeleteDaemonSet(w.client, WindowsDaemonSetName, kubeSystemNameSpace, *ownerRef); err != nil {
+		return err
+	}
+	w.MarkDeleted()
+	return nil
+}
+
+func (w *windows) MarkDeleted() {
+	w.isCreated = false
+}
+
+// RegisterPortworxPluginComponent registers the PortworxPlugin component
+func RegisterWindowsComponent() {
+	Register(WindowsComponentName, &windows{})
+}
+
+func init() {
+	RegisterWindowsComponent()
+}
+
+func (w *windows) createDaemonSet(filename string, ownerRef *metav1.OwnerReference, cluster *corev1.StorageCluster) error {
+	daemonSet, err := k8s.GetDaemonSetFromFile(filename, pxutil.SpecsBaseDir())
+	if err != nil {
+		return err
+	}
+
+	daemonSet.Name = WindowsDaemonSetName
+	daemonSet.Namespace = kubeSystemNameSpace
+	daemonSet.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+	for i, container := range daemonSet.Spec.Template.Spec.Containers {
+		var image string
+		if container.Name == "node-driver-registrar" {
+			image = w.getDesiredNodeRegistrarImage(cluster)
+
+		}
+		if container.Name == "liveness-probe" {
+			image = w.getDesiredLivenessImage(cluster)
+		}
+		daemonSet.Spec.Template.Spec.Containers[i].Image = image
+	}
+
+	existingDaemonSet := &appsv1.DaemonSet{}
+	getErr := w.client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      WindowsDaemonSetName,
+			Namespace: kubeSystemNameSpace,
+		},
+		existingDaemonSet,
+	)
+	if getErr != nil && !errors.IsNotFound(getErr) {
+		return getErr
+	}
+
+	pxutil.ApplyStorageClusterSettingsToPodSpec(cluster, &daemonSet.Spec.Template.Spec)
+
+	equal, _ := util.DeepEqualPodTemplate(&daemonSet.Spec.Template, &existingDaemonSet.Spec.Template)
+	if !equal && !w.isCreated {
+		if err := k8s.CreateOrUpdateDaemonSet(w.client, daemonSet, ownerRef); err != nil {
+			return err
+		}
+	}
+	w.isCreated = true
+	return nil
+}
+
+func (w *windows) getDesiredNodeRegistrarImage(cluster *corev1.StorageCluster) string {
+	imageName := k8s.DefaultK8SRegistryPath + "/sig-storage/csi-node-driver-registrar:v2.6.2"
```

Comment:
Let's not hard code the default image here. These should come from the csi generator. Check the csi.go for reference.

Same for the liveness image.

Reply:
done